packages/cli/src/commands/generate/cell/cell.js

```diff
@@ -61,9 +63,30 @@ export const files = async ({
     templateNameSuffix = 'List'
     // override operationName so that its find_operationName
   }
-  const operationName = await uniqueOperationName(cellName, {
-    list: shouldGenerateList,
-  })
+
+  let operationName = options.query == null ? '' : options.query.trim()
+  if (operationName == '') {
```

Comment:
I think the if statement could look something like this

```
if (options.query) {
// if a custom query name is supplied
} else {
// do the usual logic
}
```

If the operation name wasn't unique, the function `uniqueOperationName` will automatically just add "2" i.e. make sure the name is unique.